src/torrent.cpp

```diff
@@ -4368,23 +4388,26 @@ namespace {
 
 		remove_time_critical_piece(index, true);
 
-		if (is_finished()
-			&& m_state != torrent_status::finished
-			&& m_state != torrent_status::seeding)
+		if (is_downloading_state(m_state))
 		{
-			// torrent finished
-			// i.e. all the pieces we're interested in have
-			// been downloaded. Release the files (they will open
-			// in read only mode if needed)
-			finished();
-			// if we just became a seed, picker is now invalid, since it
-			// is deallocated by the torrent once it starts seeding
-		}
+			if (is_finished()
+				&& m_state != torrent_status::finished
+				&& m_state != torrent_status::seeding)
+			{
+				// torrent finished
+				// i.e. all the pieces we're interested in have
+				// been downloaded. Release the files (they will open
+				// in read only mode if needed)
+				finished();
+				// if we just became a seed, picker is now invalid, since it
+				// is deallocated by the torrent once it starts seeding
+			}
 
-		m_last_download = m_ses.session_time();
+			m_last_download = m_ses.session_time();
 
-		if (m_share_mode)
-			recalc_share_mode();
+			if (m_share_mode)
+				recalc_share_mode();
```

Comment:
all of the things above only make sense to do if we're currently downloading, not during checking. Specifically, the call to ``finished()`` would previously interrupt the checking.